x-pack/plugins/canvas/public/state/defaults.js

```diff
@@ -16,6 +16,7 @@ export const getDefaultElement = () => {
       height: 300,
       width: 500,
       angle: 0,
+      type: 'element',
```

Comment:
As noted in comment, we don't need this as these objects are already in the `elements` array, so we can divine the type from their location.

Reply:
@rashidkpc it looked like plain sailing initially but I've got 2nd thoughts, because in a bunch of locations, we concatenate `elements` and `groups` (to get `nodes` which is purely calculated, not stored). 

So it'd be simpler/cheaper to do a single `elements.concat(groups)` than having to do `elements.map(e => ({...e, type: 'element')).concat(groups.map(g => ({...g, type: 'group'})))` - storing these feels cheap in comparison to having to rebuild objects. 

A couple of algos we have now basically treat elements and groups uniformly as nodes in a tree, doing tree traversal, and it's needed to know the type. 

Alternatively I could infer from `node.id.startsWith('group')` but that feels more fragile.

Pls. give guidance if it's OK to keep the `type` around or if there's a stronger reason to remove them than the incurred downstream processing needs and code additions.

The situation is similar with `expression`, though omitting them when saving would still need object augmentation upon loading (the way it works now, it's every single drag&drop), but again, there might be some gain you have in mind. Also, it feels the `expression` would be handy to keep around, as data-driven future group behavior - eg. auto-generate a trellised plot inside itself, where the panel count would depend on the data - or automatic group sizing and positioning functions could go into the `expression`. 

What do you think, should I not worry about these now and omit these fields, or keep around for code and node structure uniformity?